process/logDataProcessor_test.go

```diff
@@ -0,0 +1,158 @@
+package process
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-ws-connector-template-go/testscommon"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewLogDataProcessor(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil marshaller, should return error", func(t *testing.T) {
+		dp, err := NewLogDataProcessor(nil)
+		require.True(t, check.IfNil(dp))
+		require.Equal(t, errNilMarshaller, err)
+	})
+
+	t.Run("should work", func(t *testing.T) {
+		dp, err := NewLogDataProcessor(&testscommon.MarshallerMock{})
+		require.False(t, check.IfNil(dp))
+		require.Nil(t, err)
+
+		err = dp.Close()
+		require.Nil(t, err)
+	})
+}
+
+func TestLogDataProcessor_ProcessPayload(t *testing.T) {
+	t.Parallel()
+
+	marshaller := &testscommon.MarshallerMock{}
+	dp, _ := NewLogDataProcessor(marshaller)
+
+	t.Run("invalid topic, should return error", func(t *testing.T) {
+		t.Parallel()
+
+		err := dp.ProcessPayload([]byte("payload"), "invalid topic")
+		require.True(t, strings.Contains(err.Error(), errInvalidOperationType.Error()))
+		require.True(t, strings.Contains(err.Error(), "payload"))
+		require.True(t, strings.Contains(err.Error(), "invalid topic"))
+	})
+
+	t.Run("save block", func(t *testing.T) {
+		t.Parallel()
+
+		err := dp.ProcessPayload([]byte("payload"), outport.TopicSaveBlock)
+		require.NotNil(t, err)
+
+		outportBlock := &outport.OutportBlock{}
+		outportBlockBytes, err := marshaller.Marshal(outportBlock)
+		require.Nil(t, err)
+
+		err = dp.ProcessPayload(outportBlockBytes, outport.TopicSaveBlock)
+		require.Nil(t, err)
```

Comment:
These tests are only half done. Can be done in this PR or another:
1. inject a logger interface on the constructor of the NewLogDataProcessor
2. using LoggerStub, you can check that the output is the desired one

Reply:
Refactored as suggested in this PR